### PR TITLE
CI updates

### PR DESCRIPTION
**Pull request by yim-lee**
- Add SwiftFormat
- Add CI for Swift nightly

### CODE REVIEW

File: Sources/NIOIMAPCore/Grammar/Date/InternalDate.swift [hunk 1]
```diff
@@ -95,7 +95,7 @@ extension ServerMessageDate {
         /// Time zone offset in minutes.
         public let zoneMinutes: Int
 
-        /// Creates a new `Components` collection from the given parameters. Note that currently no sanity checks are performed.
+        /// Creates a new `Components` collection from the given parameters. Note that currently no language checks are performed.
```
Comment (danieleggert) [inline Sources/NIOIMAPCore/Grammar/Date/InternalDate.swift:98]:
```suggestion
        /// Creates a new `Components` collection from the given parameters. Note that currently no soundness checks are performed.
```

**Review by danieleggert — COMMENTED** [review]
Can you add this
```diff
--- a/.swiftformat
+++ b/.swiftformat
@@ -1,6 +1,6 @@
 # file options
 
---swiftversion 5.4
+--swiftversion 5.7
 --exclude .build
 
 # format options
@@ -15,3 +15,4 @@
 --decimalgrouping ignore
 --binarygrouping ignore
 --octalgrouping ignore
+--extensionacl on-declarations
```

That should resolve the formatting changes. Thanks.

File: Sources/NIOIMAPCore/Grammar/Date/InternalDate.swift (hunk 1 above)
Comment (yim-lee) [reply Sources/NIOIMAPCore/Grammar/Date/InternalDate.swift:98]: Meh, GH UI doesn't let me commit this suggestion for some reason, so I applied it in my commit.

**Comment by yim-lee** [comment]
Ah I see, repo visibility changes are throwing GH off. I will open another PR.

**Pull request closed without merging**